drivers/sensor/tmp116/tmp116.c

```diff
@@ -143,8 +189,9 @@ static struct tmp116_data tmp116_data;
 
 static const struct tmp116_dev_config tmp116_config = {
 	.i2c_addr = DT_INST_REG_ADDR(0),
+	.i2c_bus_label = DT_INST_BUS_LABEL(0)
 };
 
-DEVICE_AND_API_INIT(tmp116, DT_INST_LABEL(0), tmp116_init,
+DEVICE_DT_INST_DEFINE(0, tmp116_init, device_pm_control_nop,
```

Comment:
Looks like you'll want to modify this line so that the instance number is not hard-coded.

I would suggest creating another macro like this:
```cpp
#define DEFINE_TMP116(_num) \
    DEVICE_DT_INST_DEFINE(_num, ... \
```

 and then add the following macro underneath.

```cpp
DT_INST_FOREACH_STATUS_OKAY(DEFINE_TMP116);
```

That will allow for multiple device instances. Also ensure you remove hard-coded instance numbers elsewhere.

Reply:
Done, tested it on hardware.